File: packages/ink-swift/Sources/InkGraves/Resources.swift
import Foundation

/// The weights committed with the repo (packages/ink-graves/assets),
/// bundled into this target as a resource — consumers who add the package
/// from git load the model with no downloads or file wrangling:
///
///     let model = try GravesModel(assets: parseModelAssets(bundledGravesWeights()))
public func bundledGravesWeights() throws -> Data {
    guard let url = Bundle.module.url(forResource: "graves-v2", withExtension: "bin") else {
        throw GravesError.missingBundledWeights
    }
    return try Data(contentsOf: url)
}

File: packages/ink-kotlin/ink-graves/src/main/kotlin/com/trylonghand/ink/graves/Resources.kt
package com.trylonghand.ink.graves

/**
 * The weights committed with the repo (packages/ink-graves/assets),
 * bundled into this module as a JAR resource — consumers who add the
 * package from git load the model with no downloads or file wrangling:
 *
 *     val model = GravesModel(parseModelAssets(bundledGravesWeights()))
 */
public fun bundledGravesWeights(): ByteArray {
    val stream = object {}.javaClass.getResourceAsStream("/graves-v2.bin")
        ?: throw GravesError.MissingBundledWeights()
    return stream.use { it.readBytes() }
}
